lib/modules/datasource/bitrise/index.ts

```diff
@@ -0,0 +1,138 @@
+  override readonly sourceUrlNote =
+    'The source URL is determined from the `source_code_url` field of the release object in the results.';
+
+  @cache({
+    namespace: `datasource-${BitriseDatasource.id}`,
+    key: ({ packageName, registryUrl }: GetReleasesConfig) =>
+      `${registryUrl}/${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    const parsedUrl = parseGitUrl(registryUrl);
+    if (detectPlatform(registryUrl) !== 'github') {
+      logger.warn(
+        `${parsedUrl.source} is not a supported Git hoster for this datasource`,
+      );
+      return null;
+    }
+
+    const result: ReleaseResult = {
+      releases: [],
+    };
+
+    const massagedPackageName = encodeURIComponent(packageName);
+    const baseApiURL =
+      parsedUrl.resource === 'github.com'
+        ? 'https://api.github.com'
+        : `https://${parsedUrl.resource}/api/v3`;
+    const packageUrl = joinUrlParts(
+      baseApiURL,
+      'repos',
+      parsedUrl.full_name,
+      'contents/steps',
+      massagedPackageName,
+    );
+
+    const { body: packageRaw } = await this.http.getJson(
+      packageUrl,
+      GithubContentResponse,
+    );
+
+    if (!is.array(packageRaw)) {
+      logger.warn(
+        { data: packageRaw, url: packageUrl },
+        'Got unexpected response for Bitrise package location',
+      );
+      return null;
+    }
+
+    for (const versionDir of packageRaw.filter((element) =>
+      semver.isValid(element.name),
+    )) {
+      const stepUrl = joinUrlParts(packageUrl, versionDir.name, 'step.yml');
+      // TODO use getRawFile when ready #30155
+      const { body } = await this.http.getJson(stepUrl, GithubContentResponse);
+      if (!('content' in body)) {
+        logger.warn(
+          { data: body, url: stepUrl },
+          'Got unexpected response for Bitrise step location',
+        );
+        return null;
+      }
+      if (body.encoding !== 'base64') {
+        logger.warn(
+          { data: body, url: stepUrl },
+          `Got unexpected encoding for Bitrise step location '${body.encoding}'`,
+        );
+        return null;
+      }
+
+      const content = fromBase64(body.content);
+      const { published_at, source_code_url } = parseSingleYaml(content, {
+        customSchema: BitriseStepFile,
+      });
+
+      const releaseTimestamp = is.string(published_at)
+        ? published_at
+        : published_at.toISOString();
+      result.releases.push({
+        version: versionDir.name,
+        releaseTimestamp,
+        sourceUrl: source_code_url,
+      });
+    }
+
+    // if we have no releases return null
+    if (!result.releases.length) {
+      return null;
+    }
+
+    return {
+      ...result,
+      homepage: `https://bitrise.io/integrations/steps/${packageName}`,
+    };
```

Comment:
```suggestion
    result.homepage = `https://bitrise.io/integrations/steps/${packageName}`;
    return result;
```

Isn't this more efficient, or does it need this shallow copy?

Reply:
As the objects in question are minimal in size, the impact is nearly not existing. 
Doing that with RenovateConfig in a loop would be a different story.